torchtune/modules/peft/_utils.py

```diff
@@ -362,19 +286,62 @@ def validate_missing_and_unexpected_for_lora(
         ]
     )
 
+    if full_model_state_dict_keys:
```

Comment:
Same with all the code from here on down.. the changes should be doable without any modifications to this function